File: work.php
<?php

namespace WpRefs\FixPage;

if (isset($_GET['test']) || (($_SERVER['SERVER_NAME'] ?? '') === 'localhost')) {
    ini_set('display_errors', 1);
    ini_set('display_startup_errors', 1);
    error_reporting(E_ALL);
}
/*
usage:

use function WpRefs\FixPage\fix_page_here;
use function WpRefs\FixPage\DoChangesToText1;
// $text = DoChangesToText1($sourcetitle, $text, $lang, $mdwiki_revid);
*/

include_once __DIR__ . '/src/include_files.php';

use function WpRefs\WprefText\fix_page;
use function WpRefs\TestBot\echo_test;

function get_curl(string $url): string
{
    $usr_agent = 'WikiProjectMed Translation Dashboard/1.0 (https://mdwiki.toolforge.org/; tools.mdwiki@toolforge.org)';
    $ch = curl_init();

    curl_setopt($ch, CURLOPT_URL, $url);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_USERAGENT, $usr_agent);

    curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 5);
    curl_setopt($ch, CURLOPT_TIMEOUT, 5);

    $output = curl_exec($ch);
    if ($output === FALSE) {
        echo_test("<br>cURL Error: " . curl_error($ch) . "<br>$url");
    }

    curl_close($ch);

    return $output;
}

function json_load_file($filename)
{
    if (!is_file($filename)) {
        return [];
    }

    $content = file_get_contents($filename);
    if ($content === false || $content === '') {
        return [];
    }

    return json_decode($content, true) ?: [];
}

function load_settings_new()
{
    // ---
    $url = "http://localhost:9001/api.php?get=language_settings";
    // ---
    if (($_SERVER['SERVER_NAME'] ?? '') === 'mdwiki.toolforge.org') {
        $url = "https://mdwiki.toolforge.org/api.php?get=language_settings";
        $data = get_curl($url);
    } else {
        $data = file_get_contents($url);
    }
    // ---
    if (!$data) {
        $localFile = __DIR__ . '/resources/language_settings.json';
        $data = is_file($localFile) ? file_get_contents($localFile) : '';
    }
    // ---
    $json = json_decode($data ?: '[]', true);
    if (!is_array($json)) {
        $json = ['results' => []];
    }
    // ---
    $data = $json['results'] ?? [];
    // ---
    $new = [];
    // ---
    foreach ($data as $key => $value) {
        $new[$value['lang_code']] = $value;
    }
    // ---
    return $new;
}

function fix_page_here($text, $title, $langcode, $sourcetitle, $mdwiki_revid)
{
    $setting = load_settings_new();
    // ---
    $lang_default = isset($setting[$langcode]) ? $setting[$langcode] : [];
    // ---
    // if (empty($lang_default)) { echo 'no settings for ' . $langcode; };
    // ---
    // var_export($lang_default);
    // ---
    $move_dots = isset($lang_default['move_dots']) && $lang_default['move_dots'] == 1;
    $expand = (isset($lang_default['expend']) && $lang_default['expend'] == 1) || true;
    $add_en_lang = isset($lang_default['add_en_lang']) && $lang_default['add_en_lang'] == 1;
    // ---
    $text = fix_page($text, $title, $move_dots, $expand, $add_en_lang, $langcode, $sourcetitle, $mdwiki_revid);
    // ---
    return $text;
}
// $text = DoChangesToText1($sourcetitle, $title, $text, $lang, $mdwiki_revid);

function DoChangesToText1($sourcetitle, $title, $text, $lang, $mdwiki_revid)
{
    // ---
    $newtext = fix_page_here($text, $title, $lang, $sourcetitle, $mdwiki_revid);
    // ---
    if (empty($newtext)) {
        $newtext = $text;
    }
    // ---
    return $newtext;
}

File: wprefs/work.py
"""Python translation of ``work.php``."""

from __future__ import annotations

import json
import os
from pathlib import Path
from urllib.error import URLError
from urllib.request import Request, urlopen

from .src import include_files
from .src.index import fix_page
from .TestBot import echo_test


include_files.include_all()

USER_AGENT = "WikiProjectMed Translation Dashboard/1.0 (https://mdwiki.toolforge.org/; tools.mdwiki@toolforge.org)"


def get_curl(url: str) -> str:
    request = Request(url, headers={"User-Agent": USER_AGENT})
    try:
        with urlopen(request, timeout=5) as response:  # nosec: B310
            return response.read().decode("utf-8")
    except URLError as exc:  # pragma: no cover - network failures
        echo_test(f"\ncURL Error: {exc}\n{url}")
        return ""


def json_load_file(filename: str):
    path = Path(filename)
    if not path.exists():
        return None
    try:
        return json.loads(path.read_text(encoding="utf-8"))
    except json.JSONDecodeError:
        return None


def load_settings_new() -> dict:
    server_name = os.environ.get("SERVER_NAME", "")
    if server_name == "mdwiki.toolforge.org":
        json_data = get_curl("https://mdwiki.toolforge.org/api.php?get=language_settings")
    else:
        json_data = get_curl("http://localhost:9001/api.php?get=language_settings")
    try:
        data = json.loads(json_data)
    except json.JSONDecodeError:
        data = {}
    results = data.get("results", [])
    new = {}
    for value in results:
        lang_code = value.get("lang_code")
        if lang_code:
            new[lang_code] = value
    return new


def fix_page_here(text: str, title: str, langcode: str, sourcetitle: str, mdwiki_revid) -> str:
    setting = load_settings_new()
    lang_default = setting.get(langcode, {})
    move_dots = bool(int(lang_default.get("move_dots", 0)))
    expand = bool(int(lang_default.get("expend", 1))) or True
    add_en_lang = bool(int(lang_default.get("add_en_lang", 0)))
    return fix_page(text, title, move_dots, expand, add_en_lang, langcode, sourcetitle, mdwiki_revid)


def DoChangesToText1(sourcetitle: str, title: str, text: str, lang: str, mdwiki_revid) -> str:
    newtext = fix_page_here(text, title, lang, sourcetitle, mdwiki_revid)
    return newtext if newtext else text
